### PR TITLE
Integration test framework and basic tests

### PR DESCRIPTION
**Pull request by kelvinatsplunk**
This PR
- adds the test framework using Ginkgo. It is under /test/testenv (Please see /test/README doc for more details).
- adds 2 basic tests using the test framework. The tests deploy a standalone (s2) and clustered (c3 - 3 indexers, 3 shs) and verified they are deployed and "ready".
- adds test scripts under /test to deploy kind or eks cluster and run the tests (Please see README doc under /test for details).
- updates Makefile and circleci's config.yaml to deploy the cluster and run the tests in the pipeline.

### CODE REVIEW

File: test/deploy-eks-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      return 1
+    fi
+  else
+    echo "Retrieving kubeconfig for ${CLUSTER_NAME}"
+    # Cluster exists but kubeconfig may not
+    eksctl utils write-kubeconfig --cluster=${CLUSTER_NAME}
+  fi
+
+  echo "Logging in to ECR"
+  rc=$(aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin 667741767953.dkr.ecr.us-west-2.amazonaws.com/splunk/splunk-operator)
```
Comment (mikedickey) [inline test/deploy-eks-cluster.sh:42]: Account info should probably come from ENV

File: test/deploy-eks-cluster.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  if [ -z "$rc" ]; then
+    echo "eksctl is not installed or in the PATH. Installing..."
+    os=$(uname -s)
+    curl --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_${os}_amd64.tar.gz" | tar xz -C /tmp
+    if [ $? -ne 0 ]; then
+     echo "Unable to download eksctl for ${os}."
+     return 1
+    fi
+    chmod +x /tmp/eksctl
+    sudo mv /tmp/eksctl /usr/local/bin
```
Comment (mikedickey) [inline test/deploy-eks-cluster.sh:25]: Not sure we should have createCluster() install this versus having it as a dependency for testing on eks clusters?

File: test/deploy-kind-cluster.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  if [ -z "$rc" ]; then
+    echo "Kind is not installed or in the PATH. Installing..."
+    os=$(uname)
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.7.0/kind-${os}-amd64
+    if [ $? -ne 0 ]; then
+     echo "Unable to download kind for ${os}."
+     return 1
+    fi
+    chmod +x ./kind
+    sudo mv kind /usr/local/bin
```
Comment (mikedickey) [inline test/deploy-kind-cluster.sh:34]: Same as eksctl - not sure about installing these things via our scripts, especially the "sudo" part of it

File: test/README.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+The test suite file to use to create and delete the TestEnv object. The test spec files are used to create and delete the Deployment object.
+
+### Adding a new test spec
+
+1. To add a new test spec, you can either add a new "It" spec in an existing test spec file or add a new test spec file. If you are adding a new test spec file,
+it is best to copy an existing test spec file in that suite and modified it.
+
+### Add a new test suite
+
+1. If you are adding a new test suite (ie you want to run the test in a separate TestEnv or k8s namespace), it is best to copy the test/example folder entirely and modified it
```
Comment (mikedickey) [inline test/README.md:89]: "and modify it"

File: test/env.sh [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+: "${COMMIT_HASH:=}"
+
+# Docker registry to use to push the test images to and pull from in the cluster
+if [ -z "${PRIVATE_REGISTRY}" ]; then
+    case ${CLUSTER_PROVIDER} in
+      kind)
+        PRIVATE_REGISTRY=localhost:5000
+        ;;
+      eks)
+        PRIVATE_REGISTRY=667741767953.dkr.ecr.us-west-2.amazonaws.com
```
Comment (mikedickey) [inline test/env.sh:18]: Think this account info should be provided via CircleCI env (not included in code)

File: test/run-tests.sh [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# Install the CRDs
+echo "Installing enterprise CRDs..."
+kubectl apply -f ${topdir}/deploy/crds
+if [ $? -ne 0 ]; then
+  echo "Unable to install the CRDs. Exiting..."
+  exit 1
+fi
+
+rc=$(which go)
+if [ -z "$rc" ]; then
```
Comment (mikedickey) [inline test/run-tests.sh:62]: might want to move these dependency checks to top, to fail fast

File: test/deploy-eks-cluster.sh (hunk 1 above)
Comment (kelvinatsplunk) [reply test/deploy-eks-cluster.sh]: Updated to use PRIVATE_REGISTRY

File: test/deploy-kind-cluster.sh (hunk 3 above)
Comment (kelvinatsplunk) [reply test/deploy-kind-cluster.sh]: Move to install this in circle config.yml

File: test/env.sh (hunk 5 above)
Comment (kelvinatsplunk) [reply test/env.sh]: Define in ECR_REGISTRY circleci env

File: test/deploy-eks-cluster.sh (hunk 2 above)
Comment (kelvinatsplunk) [reply test/deploy-eks-cluster.sh]: Agree. Install this in circleci config.yml

File: test/run-tests.sh (hunk 6 above)
Comment (kelvinatsplunk) [reply test/run-tests.sh]: Good catch!

**Pull request merged**